gulpHelpers.js

```diff
@@ -68,39 +71,71 @@ module.exports = {
       }
     });
 
+    Object.keys(libraries).forEach(library => {
+      if (!modules.includes(library) && modules.some(module => libraries[library].dependants.includes(module))) {
+        modules.unshift(library);
+      }
+    });
+
     return modules;
   },
+  getParentLibraries(moduleName) {
+    const libraryNames = [];
+    Object.keys(libraries).forEach(libraryName => {
+      const library = libraries[libraryName];
+      if (library.dependants.includes(moduleName)) {
+        libraryNames.push(libraryName);
+      }
+    });
+    return libraryNames;
+  },
+  getLibraryFiles(name) {
+    const library = libraries[name];
+    const files = library.files.map(file => require.resolve(file, {paths: ['./libraries/' + name + '/']}));
+    return files;
+  },
   getModules: _.memoize(function(externalModules) {
     externalModules = externalModules || [];
     var internalModules;
     try {
+      var getInternalModules = function(absolutePath) {
+        return fs.readdirSync(absolutePath)
+          .filter(file => (/^[^\.]+(\.js)?$/).test(file))
+          .reduce((memo, file) => {
+            var moduleName = file.split(new RegExp('[.\\' + path.sep + ']'))[0];
+            var modulePath = path.join(absolutePath, file);
+            if (fs.lstatSync(modulePath).isDirectory()) {
+              modulePath = path.join(modulePath, 'index.js')
+            }
+            if (fs.existsSync(modulePath)) {
+              memo[modulePath] = moduleName;
+            }
+            return memo;
+          }, {});
+      };
+
       var absoluteModulePath = path.join(__dirname, MODULE_PATH);
-      internalModules = fs.readdirSync(absoluteModulePath)
-        .filter(file => (/^[^\.]+(\.js)?$/).test(file))
-        .reduce((memo, file) => {
-          var moduleName = file.split(new RegExp('[.\\' + path.sep + ']'))[0];
-          var modulePath = path.join(absoluteModulePath, file);
-          if (fs.lstatSync(modulePath).isDirectory()) {
-            modulePath = path.join(modulePath, 'index.js')
-          }
-          if (fs.existsSync(modulePath)) {
-            memo[modulePath] = moduleName;
-          }
-          return memo;
-        }, {});
+      var absoluteLibraryPath = path.join(__dirname, LIBRARY_PATH);
+
+      internalModules = getInternalModules(absoluteModulePath);
+      var internalLibraries = getInternalModules(absoluteLibraryPath);
```

Comment:
I don't think this is the right logic to use for libraries - because the dependencies on them will not work unless they are explicitly declared in `.submodules.json`, I think this should just return the list of libraries as it's defined in there.
Otherwise this allows inconsistencies when the library does not use `index.js` in the way we expect here, or the  compilation of files that are not used by anyone.